api/v1alpha1/spire_server_config_types.go

```diff
@@ -149,6 +151,14 @@ type DataStore struct {
 	ClientCertPath string `json:"clientCertPath,omitempty"`
 	ClientKeyPath  string `json:"clientKeyPath,omitempty"`
 
+	// tlsSecretName specifies the name of a Kubernetes Secret containing TLS certificates for database connections.
+	// The Secret will be mounted at /run/spire/db-tls in the SPIRE server container.
+	// The Secret should contain keys like 'ca.crt', 'tls.crt', 'tls.key' for the respective certificates.
+	// For PostgreSQL, reference these certificates in the connectionString, e.g.:
+	// "sslmode=verify-full sslrootcert=/run/spire/db-tls/ca.crt sslcert=/run/spire/db-tls/tls.crt sslkey=/run/spire/db-tls/tls.key"
+	// +kubebuilder:validation:Optional
+	TLSSecretName string `json:"tlsSecretName,omitempty"`
```

Comment:
spire server is not kube-native, hence, it does not load from secrets. The username and password used in the connection string needs to be secured. For now, the connection string can be considered as input from the user on the CR. In the future, the username and password can be set as keywords, that will enable injection without adding the details to configmap.